src/compiler/scanner.ts

```diff
@@ -2763,25 +2794,26 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
 
         function scanPatternModifiers(currFlags: RegularExpressionFlags): RegularExpressionFlags {
             while (true) {
-                const ch = charCodeChecked(pos);
+                const ch = codePointChecked(pos);

```

Comment:
Should this be moved to #58612 as well?

Reply:
Ah, yes. It was a mistake that I thought I did entirely reverted.